pkg/reconciler/instances/istio/ingress-gateway/deployment.go

```diff
@@ -0,0 +1,59 @@
+package ingressgateway
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	istioCR "github.com/kyma-project/istio/operator/api/v1alpha1"
+)
+
+const (
+	namespace      string = "istio-system"
+	name           string = "istio-ingressgateway"
+	annotationName string = "reconciler.kyma-project.io/lastRestartDate"
+)
+
+func RestartDeployment(ctx context.Context, k8sClient client.Client) error {
+	deployment := appsv1.Deployment{}
+	err := k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, &deployment)
+	if err != nil {
+		return err
+	}
+
+	if len(deployment.Spec.Template.Annotations) == 0 {
+		deployment.Spec.Template.Annotations = make(map[string]string)
+	}
+	deployment.Spec.Template.Annotations[annotationName] = time.Now().Format(time.RFC3339)
+
+	return k8sClient.Update(ctx, &deployment)
+}
+
+func NeedsRestart(ctx context.Context, client client.Client, istioCRList *istioCR.IstioList) (bool, error) {
+	numTrustedProxies, err := GetNumTrustedProxyFromIstioCM(ctx, client)
+	if err != nil {
+		return false, err
+	}
+
+	// Restart to default if no IstioCR is present
+	if numTrustedProxies != nil && len(istioCRList.Items) == 0 {
```

Comment:
can we have a test case for that? I would like to check this len of istioCRList

Reply:
https://github.com/barchw/reconciler/blob/098a8d3acec6b011a679c12caa4ed13bc3f73b69/pkg/reconciler/instances/istio/ingress-gateway/deployment_test.go#L72

Like this?

Reply:
```go
func TestIngressGatewayNeedsRestartNoIstioCr(t *testing.T) {
	t.Run("should not restart when there's no Istio CR, and CM has no configuration for numTrustedProxies", func(t *testing.T) {
		client := GetClientSet(t, TestConfigMapEmpty)
		does, err := ingressgateway.NeedsRestart(context.TODO(), client, &istioCR.IstioList{Items: []istioCR.Istio{}})

		require.NoError(t, err)
		require.False(t, does)
	})

	t.Run("should restart when there's no Istio CR, and CM has configuration for numTrustedProxies", func(t *testing.T) {
		client := GetClientSet(t, TestConfigMap)
		does, err := ingressgateway.NeedsRestart(context.TODO(), client, &istioCR.IstioList{Items: []istioCR.Istio{}})

		require.NoError(t, err)
		require.True(t, does)
	})
}
```

There's this one, do you mean some other?

Reply:
Right! this is the one, I think comment mislead me a bit